### PR TITLE
DSND-2433: Transform composite resolutions

### PR DESCRIPTION
**Pull request by cgregoryCH**
## Describe the changes
* Handles transformation of composite resolutions.
* Refactor pre-transform mapping and child transformation logic
* All pre-transform mapping is done before transform, including children.
* Transformation of children is handled by checking if each of the child arrays exist.
* RES15 not included as it should be handled differently.

### Related Jira tickets
[DSND-2433](https://companieshouse.atlassian.net/browse/DSND-2433)

### General
- [ ] Is the PR as small as it can be?
- [ ] Has the Jira ticket been updated with any relevant comments?
- [ ] Is the `README` up to date?
- [ ] Has the `POM` been updated for the latest versions of dependencies?
- [ ] Has the code been double-checked against `main`?
- [ ] Does the code adhere standards in this repository, including SonarQube checks?

### Testing
- [ ] Do the code changes have unit and integration tests with code coverage > 80%?
- [ ] Has the code been tested locally and/or passed the API Karate tests on Tilt?
- [ ] Have mandatory manual test cases been run?
- [ ] Are extra Karate tests required?
- [ ] Are all the test data resources up to date?

### Release preparation
_Where possible, add links to the respective Jira tickets when an item is checked_
- [ ] Have these changes been included in a release ticket?
- [ ] Are changes required to the `docker-chs-development` deployment or test data?
- [ ] Are any changes required to the environment added to `chs-configs`?

## Notes to the tester
_Add any testing hints or instructions here._

[DSND-2433]: https://companieshouse.atlassian.net/browse/DSND-2433?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by sthompsonCH — APPROVED** [review]

**Review by crowlandsCH — APPROVED** [review]

File: src/test/java/uk/gov/companieshouse/filinghistory/consumer/mapper/pretransform/AssociatedFilingNodeMapperTest.java [hunk 1]
```diff
@@ -6,53 +6,58 @@
...
+    private AssociatedFilingNodeMapper associatedFilingNodeMapper;
     @Mock
     private FormatDate formatDate;
 
     @Mock
     private FilingHistory delta;
 
+    @BeforeEach
+    void setUp() {
+        associatedFilingNodeMapper = new AssociatedFilingNodeMapper(objectMapper, formatDate);
```
Comment (johncook-ch) [inline src/test/java/uk/gov/companieshouse/filinghistory/consumer/mapper/pretransform/AssociatedFilingNodeMapperTest.java:33]: Do we need this? The value can be assigned once in the declaration, assuming the mapper is stateless

File: src/test/java/uk/gov/companieshouse/filinghistory/consumer/mapper/pretransform/DeltaNodeMapperTest.java [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+
+    private static final ObjectMapper objectMapper = TransformerTestingUtils.getMapper();
+
+    private DeltaNodeMapper preTransformMapper;
+    @Mock
+    private FormatDate formatDate;
+
+    @BeforeEach
+    void setUp() {
+        preTransformMapper = new DeltaNodeMapper(objectMapper, formatDate);
```
Comment (johncook-ch) [inline src/test/java/uk/gov/companieshouse/filinghistory/consumer/mapper/pretransform/DeltaNodeMapperTest.java:34]: Do we need this? The value can be assigned once in the declaration, assuming the mapper is stateless

File: src/test/java/uk/gov/companieshouse/filinghistory/consumer/mapper/pretransform/EmbeddedChildNodeMapperTest.java [hunk 3]
```diff
@@ -8,56 +8,60 @@
...
 
-    @Mock
-    private ObjectMapper objectMapper;
+    private EmbeddedChildNodeMapper embeddedChildNodeMapper;
     @Mock
     private FormatDate formatDate;
 
+    @BeforeEach
+    void setUp() {
+        embeddedChildNodeMapper = new EmbeddedChildNodeMapper(objectMapper, formatDate);
```
Comment (johncook-ch) [inline src/test/java/uk/gov/companieshouse/filinghistory/consumer/mapper/pretransform/EmbeddedChildNodeMapperTest.java:33]: Do we need this? The value can be assigned once in the declaration, assuming the mapper is stateless

File: src/test/java/uk/gov/companieshouse/filinghistory/consumer/mapper/pretransform/ResolutionNodeMapperTest.java [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    private ResolutionNodeMapper resolutionNodeMapper;
+    @Mock
+    private FormatDate formatDate;
+
+    @Mock
+    private FilingHistory delta;
+
+    @BeforeEach
+    void setUp() {
+        resolutionNodeMapper = new ResolutionNodeMapper(objectMapper, formatDate);
```
Comment (johncook-ch) [inline src/test/java/uk/gov/companieshouse/filinghistory/consumer/mapper/pretransform/ResolutionNodeMapperTest.java:41]: Do we need this? The value can be assigned once in the declaration, assuming the mapper is stateless

**Review by johncook-ch — CHANGES_REQUESTED** [review]

File: src/test/java/uk/gov/companieshouse/filinghistory/consumer/mapper/pretransform/AssociatedFilingNodeMapperTest.java (hunk 1 above)
Comment (johncook-ch) [reply src/test/java/uk/gov/companieshouse/filinghistory/consumer/mapper/pretransform/AssociatedFilingNodeMapperTest.java:33]: @cgregoryCH explained that this fixes a mock ordering issue

File: src/test/java/uk/gov/companieshouse/filinghistory/consumer/mapper/pretransform/EmbeddedChildNodeMapperTest.java (hunk 3 above)
Comment (johncook-ch) [reply src/test/java/uk/gov/companieshouse/filinghistory/consumer/mapper/pretransform/EmbeddedChildNodeMapperTest.java:33]: @cgregoryCH explained that this fixes a mock ordering issue

File: src/test/java/uk/gov/companieshouse/filinghistory/consumer/mapper/pretransform/ResolutionNodeMapperTest.java (hunk 4 above)
Comment (johncook-ch) [reply src/test/java/uk/gov/companieshouse/filinghistory/consumer/mapper/pretransform/ResolutionNodeMapperTest.java:41]: @cgregoryCH explained that this fixes a mock ordering issue

File: src/test/java/uk/gov/companieshouse/filinghistory/consumer/mapper/pretransform/DeltaNodeMapperTest.java (hunk 2 above)
Comment (johncook-ch) [reply src/test/java/uk/gov/companieshouse/filinghistory/consumer/mapper/pretransform/DeltaNodeMapperTest.java:34]: @cgregoryCH explained that this fixes a mock ordering issue

**Review by johncook-ch — APPROVED** [review]

**Pull request merged**